opendevin/runtime/plugins/agent_skills/agentskills.py

```diff
@@ -41,30 +41,84 @@
-OPENAI_MODEL = os.getenv('OPENAI_MODEL', 'gpt-4o-2024-05-13')
-MAX_TOKEN = os.getenv('MAX_TOKEN', 500)
+# TODO: Move this to EventStream Actions when EventStreamRuntime is fully implemented
+# NOTE: we need to get env vars inside functions because they will be set in IPython
+# AFTER the agentskills is imported (the case for EventStreamRuntime)
+# ==================================================================================================
+def _get_openai_api_key():
+    return os.getenv('OPENAI_API_KEY', os.getenv('SANDBOX_ENV_OPENAI_API_KEY', ''))
+
+
+def _get_openai_base_url():
+    return os.getenv('OPENAI_BASE_URL', 'https://api.openai.com/v1')
+
+
+def _get_openai_model():
+    return os.getenv('OPENAI_MODEL', 'gpt-4o-2024-05-13')
+
 
-OPENAI_PROXY = f'{OPENAI_BASE_URL}/chat/completions'
+def _get_max_token():
+    return os.getenv('MAX_TOKEN', 500)
 
-client = OpenAI(api_key=OPENAI_API_KEY, base_url=OPENAI_BASE_URL)
+
+def _get_openai_client():
+    client = OpenAI(api_key=_get_openai_api_key(), base_url=_get_openai_base_url())
+    return client
+
+
+# ==================================================================================================
 
 
 # Define the decorator using the functionality of UpdatePwd
 def update_pwd_decorator(func):
     @functools.wraps(func)
     def wrapper(*args, **kwargs):
-        old_pwd = os.getcwd()
         jupyter_pwd = os.environ.get('JUPYTER_PWD', None)
+        try:
+            old_pwd = os.getcwd()
+        except FileNotFoundError:
+            import json
+            import subprocess
+
+            print(
+                f'DEBUGGING Environment variables: {json.dumps(dict(os.environ), indent=2)}'
+            )
+            print(f'DEBUGGING User ID: {os.getuid()}, Group ID: {os.getgid()}')
+            import sys
+
+            print(f'DEBUGGING Loaded modules: {list(sys.modules.keys())}')
+            import tempfile
+
+            try:
+                tempfile.TemporaryFile(dir=os.getcwd())
+                print(f'DEBUGGING Directory {os.getcwd()} is writable')
+            except Exception as e:
+                print(f'DEBUGGING Directory {os.getcwd()} is not writable: {str(e)}')
+
+            out = subprocess.run(['pwd'], capture_output=True)
+            old_pwd = out.stdout.decode('utf-8').strip()
+            print(f'DEBUGGING OLD working directory: {old_pwd}')
+
+            os.chdir(old_pwd)
+            print(f'DEBUGGING change to OLD working directory: {old_pwd}')
+
+            # ls -alh
+            out = subprocess.run(['ls', '-alh', '$(pwd)'], capture_output=True)
+            print(
+                f'DEBUGGING OLD working directory contents: {out.stdout.decode("utf-8")}'
+            )
+
+            # whoami
+            out = subprocess.run(['whoami'], capture_output=True)
+            print(f'DEBUGGING OLD whoami: {out.stdout.decode("utf-8")}')
+            print(f'DEBUGGING JUPYTER pwd: {jupyter_pwd}')
```

Comment:
a lot of `print`s 😅 , intended?

Reply:
the CI was broken, these was added to desperately debug the issue 😢